src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,128 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.comparingInt;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream()
+                .map(Employee::getPerson)
+                .mapToInt(Person::getAge)
+                .average().getAsDouble();
+
+        assertThat(expected, Matchers.closeTo(33.66, 0.1));
+    }
+
+    @Test
+    void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Comparator<Person> comparator = comparingInt(person -> person.getFullName().length());
+
+        Person expected = employees.stream()
+                .map(Employee::getPerson)
+                .max(comparator).get();
+
+        assertThat(expected, Matchers.is(employees.get(1).getPerson()));
+    }
+
+    @Test
+    void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        Employee expected = employees.stream()
+                .collect(Collectors.toMap(Function.identity(), employee -> employee.getJobHistory()
+                        .stream()
+                        .collect(Collectors.toMap(JobHistoryEntry::getPosition, JobHistoryEntry::getDuration,
+                                (duration1, duration2) -> duration1 + duration2))
+                        .entrySet().stream()
+                        .max(comparing(Map.Entry::getValue)).get().getValue()
+                ))
+                .entrySet().stream()
+                .max(comparingInt(Map.Entry::getValue)).get().getKey();
+
```

Comment:
You can rewrite your code like that to make it more readable:
```
.collect(toMap(Function.identity(), employee -> employee.getJobHistory()
                                                        .stream()
                                                        .collect(toMap(JobHistoryEntry::getPosition, JobHistoryEntry::getDuration, Integer::sum))
                                                        .entrySet()
                                                        .stream()
                                                        .map(Map.Entry::getValue)
                                                        .max(Integer::compareTo)
                                                        .get()
))
.entrySet()
.stream()
.max(comparingInt(Map.Entry::getValue))
.get()
.getKey();
```
I didn't change the second part of that stream, because in this case it will be like in my HW (and you can check it here: https://github.com/elefus/epam-lab-java8-2018-09/pull/364/files )